### PR TITLE
Implement clusterctl and commit vendor directory.

### PR DESCRIPTION
**Pull request by davidewatson**

### CODE REVIEW

**Comment by alejandroEsc** [comment]
I would suggest you dont commit vendor directory for now.

File: cloud/ssh/actuators/machine/actuator.go [hunk 1]
```diff
@@ -17,10 +17,23 @@ import (
 	"fmt"
 
 	"github.com/golang/glog"
+	clustercommon "sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
 	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 	client "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
 )
 
+const (
+	ProviderName = "metal"
```
Comment (alejandroEsc) [inline cloud/ssh/actuators/machine/actuator.go]: why metal? hmmmm..could we just go with SSH?

**Comment by alejandroEsc** [comment]
LGTM, just one question, after i can merge.

**Comment by davidewatson** [comment]
I didn't change it.

**Pull request merged**